quests.md

```diff
@@ -0,0 +1,272 @@
+
+Keep in mind that all values will be accepted, but not all will be valid, even values in this list
+```
+enum ConditionType {
+    UNSET = 0;
+    WITH_POKEMON_TYPE = 1;
+    WITH_POKEMON_CATEGORY = 2;
+    WITH_WEATHER_BOOST = 3;
+    WITH_DAILY_CAPTURE_BONUS = 4;
+    WITH_DAILY_SPIN_BONUS = 5;
+    WITH_WIN_RAID_STATUS = 6;
+    WITH_RAID_LEVEL = 7;
+    WITH_THROW_TYPE = 8;
+    WITH_WIN_GYM_BATTLE_STATUS = 9;
+    WITH_SUPER_EFFECTIVE_CHARGE = 10;
+    WITH_ITEM = 11;
+    WITH_UNIQUE_POKESTOP = 12;
+    WITH_QUEST_CONTEXT = 13;
+    WITH_THROW_TYPE_IN_A_ROW = 14;
+    WITH_CURVE_BALL = 15;
+    WITH_BADGE_TYPE = 16;
+    WITH_PLAYER_LEVEL = 17;
+    WITH_WIN_BATTLE_STATUS = 18;
+    WITH_NEW_FRIEND = 19;
+    WITH_DAYS_IN_A_ROW = 20;
+}
+```
+
+### Conditions Detail
+When the basic fields of the condition don't cover all use-cases, reference below
+
+Some possibilities from the conditions type array were left out because they didn't seem useful to have any detail on
+
+#### Pokemon Type (type `1`)
+See [PA's type array](https://github.com/PokeAlarm/PokeAlarm/blob/master/locales/en.json#L1016) for a complete type
+id/name correlation (minus the padded zeroes)
+```json
+{
+  "type_id": [int]
+}
+```
+
+#### Pokemon Category (type `2`)
+How they define a group of monsters without any particular relation
+```json
+{
+  "pokemon_id": [int],
+  "name": string
+}
+```
+* The `name` field is optional, and probably won't ever be filled
+
+#### Raid Level (type `7`)
+```json
+{
+  "level": [1-5]
+}
+```
+
+#### Throw Type (type `8`)
+```json
+{
+  "throw_id": int
+}
+```
+This can have one of the following
+
+| ID | Name      |
+|----|-----------|
+| 0  | Normal    |
+| 1  | Nice      |
+| 2  | Great     |
+| 3  | Excellent |
+
+For `WITH_THROW_TYPE_IN_A_ROW`, increase the `amount` field on the condition
+
+#### Item (type `11`)
+```json
+{
+  "item_id": int
+}
+```
+See [item array](#item-array) for id options
+
+
+
+
+
+## Rewards
+
+This spec will cover most of the requirements, but when more information is needed, the `detail` field should be 
+included
+
+Otherwise, the `detail` field will not be checked
+
+| Key           | Type   | Description                                                                           |
+|---------------|--------|---------------------------------------------------------------------------------------|
+| type          | int    | See [the reward types array](#rewards-type-array) for options                         |
+| id            | int    | A generic id representing an object, based on `type`                                  |
+| amount        | int    | A generic amount representing how many must be acquired or completed, based on `type` |
+| detail        | object | See [the rewards detail section](#rewards-detail) for options                         |
+
+
+```json
+{
+  "type": int,
+  "id": int,
+  "amount": int,
+  "detail": {...}
```

Comment:
Why nest in detail? Why not just have optional fields within the reward object

Reply:
I would really like this to be called `info` as already implemented by RM++ and RDM